### PR TITLE
blockchain_db: remove unused get_output_key variant

### PR DESCRIPTION
**Pull request by moneromooo-monero**
It was actually incorrect, as it would not return commitment

### CODE REVIEW

File: src/blockchain_db/berkeleydb/db_bdb.h [hunk 1]
```diff
@@ -419,6 +418,7 @@ class BlockchainBDB : public BlockchainDB
    * @return the global index of the desired output
    */
   uint64_t get_output_global_index(const uint64_t& amount, const uint64_t& index);
+  output_data_t get_output_key(const uint64_t& global_index) const;
```
[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply src/blockchain_db/berkeleydb/db_bdb.h]: It's not a vrtual function anymore, so I moved it there.

**Comment by moneromooo-monero** [comment]
Not as far as I know. That's only one function that gets outputs.

File: src/blockchain_db/berkeleydb/db_bdb.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (stoffu) [reply src/blockchain_db/berkeleydb/db_bdb.h]: Why not simply delete the function body in cpp, just like with LMDB?

[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply src/blockchain_db/berkeleydb/db_bdb.h]: The berkeley db class is obsolete, so I didn't actually care. I think this whole class can be removed tbh. Its only point now is to ensure the "pluggability" still works.

[the remark this answers is not in the document]
Comment (stoffu) [reply src/blockchain_db/berkeleydb/db_bdb.h]: OK

**Pull request merged**